rust/arrow/src/array/array_primitive.rs

```diff
@@ -67,19 +67,6 @@ impl<T: ArrowPrimitiveType> PrimitiveArray<T> {
         self.data.is_empty()
     }
 
-    /// Returns a slice for the given offset and length
```

Comment:
Since this is a breaking change in the API, it might be worth doing as part of a separate PR. What do you think @jorgecarleitao and @nevi-me ?

Reply:
My rationale was that this would have already been removed as part of ARROW-10989, if not for this one usage in the comparison kernels. The straight-forward replacement would be `values()[offset..offset+len]`.

Reply:
I missed this one, sorry.

I am fine removing it altogether. The change of signature from `safe` to `unsafe` in a previous commit already made this backward incompatible, anyways. As a user, I will prefer to migrate it to `values()[offset..offset+len]` instead of introducing an `unsafe` in my code base.

Reply:
The last merge reintroduced the method, I created a followup ticket to remove it and also from `BooleanArray` so we have the api change documented in a ticket.

Reply:
I see it https://issues.apache.org/jira/browse/ARROW-11099 👍  thanks @jhorstmann